python/python/lance/dataset.py

```diff
@@ -1914,8 +1920,21 @@ def offset(self, n: Optional[int] = None) -> ScannerBuilder:
         self._offset = n
         return self
 
-    def columns(self, cols: Optional[list[str]] = None) -> ScannerBuilder:
-        self._columns = cols
+    def columns(
+        self, cols: Optional[Union[List[str], Dict[str, str]]] = None
+    ) -> ScannerBuilder:
+        if cols is None:
+            self._columns = None
+        elif isinstance(cols, dict):
+            self._columns_with_transform = [
+                (name, expr) for (name, expr) in cols.items()
+            ]
```

Comment:
```suggestion
            self._columns_with_transform = cols.items()
```
seems it could be simplified

Reply:
i'll make it `list(cols.items())`. still need to make sure it's a list